tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7624,6 +7624,9 @@ def warn_hlsl_narrowing : Warning<
 def warn_hlsl_sema_minprecision_promotion : Warning <
   "%0 is promoted to %1">,
   InGroup<Conversion>, DefaultWarn;
+def warn_hlsl_ambiguous_literal_shift : Warning <
+  "use a type suffix on literal values, like 'L' or 'U', or a cast, to resolve ambiguity">,
+  InGroup<Conversion>, DefaultWarn;
```

Comment:
Also, here we need to change the group from Conversion to a new custom group, like:
```
def HLSLAmbigLitShift : DiagGroup<"ambig-lit-shift">;
```
just after HLSLCommaInInit in DiagnosticGroups.td.

That way it can be disabled independently from other Conversion warnings.